src/adagrams.js

```diff
@@ -1,15 +1,129 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+export default class Adagrams {
+  constructor() {
+    // this.score = 0;
+    // this.hand = [];
+  }
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+  drawLetters() {
+    const letterPool = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    };
+    const letters = Object.keys(letterPool);
+    let hand = [];
+    let i = 0;
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+    while (i < 10) {
```

Comment:
This loops makes sense, and I don't think you really have to worry about it here, but when I see while loops like this, I think about adding some other kind of guard condition- like limiting the number of attempts, for example-- to catch a possible infinite loop if there weren't enough letters to be picked.